core/services/fluxmonitor/flux_monitor.go

```diff
@@ -576,6 +576,12 @@ func (p *PollingDeviationChecker) respondToNewRoundLog(log *contracts.LogNewRoun
 	if !roundState.EligibleToSubmit {
 		logger.Infow("Ignoring new round request: not eligible to submit", p.loggerFieldsForNewRound(log)...)
 		return
+	} else if roundState.AvailableFunds.Cmp(roundState.PaymentAmount) < 0 {
+		logger.Infow("Ignoring new round request: aggregator is underfunded", p.loggerFieldsForNewRound(log)...)
+		return
+	} else if roundState.PaymentAmount.Cmp(p.store.Config.MinimumContractPayment().ToInt()) < 0 {
+		logger.Infow("Ignoring new round request: round payment amount < minimum contract payment", p.loggerFieldsForNewRound(log)...)
+		return
```

Comment:
It feels like there's a lot of duplication between this block and [this one](https://github.com/smartcontractkit/chainlink/pull/2559/files#diff-419f86f11959df769caae454a90c50ccR638). Any way we can combine some of this?